templates/cpp-template-default/proj.android/app/build.gradle

```diff
@@ -111,22 +110,24 @@ android {
 }
 
 android.applicationVariants.all { variant ->
-    def project_root_folder = "${projectDir}/../.."
-    def dest_assets_folder = "${projectDir}/assets"
-
-    // delete previous files first
-    delete dest_assets_folder
-    def targetName = variant.name.capitalize()
-    def copyTaskName = "copy${targetName}ResourcesToAssets"
-    
-    tasks.register(copyTaskName) {
+    def mergeAssetsTask = variant.mergeAssetsProvider.get()
+    def projectRootFolder = "${projectDir}/../.."
+    def resourcesDir = "${projectRootFolder}/Resources"
+    def assetsOutputDir = mergeAssetsTask.outputDir.get()
+
+    mergeAssetsTask.doFirst {
+        // delete previous files first
+        delete "${assetsOutputDir}/*"
+    }
+
+    mergeAssetsTask.doLast {
```

Comment:
As @PatriceJiang told me that, `doLast` may not be invoked.

Reply:
I was not aware of that behavior of `doLast`, so it's best to close this PR, since the fix added by @PatriceJiang works fine as it is.